scripts/verify-replace.sh

```diff
@@ -16,26 +19,61 @@ do
+    stdout_file=$(mktemp ${BASE_REPO_PATH}/$(basename ${repo})-output.XXX)
+    echo "std output file : ${stdout_file}"
     echo "Cloning repo in /tmp"
     git clone --depth=1 ${repo} ${repo_path}
     echo "Repo cloned successfully"
     cd ${repo_path}
     if ! make pre-verify; then
-        ERRORLIST+="($(basename ${repo}))"
+        ERRORREPOLIST+="$(basename ${repo}) "
         continue
     fi
     echo "Initiating 'go mod replace' of current toolchain common version in dependent repos"
     go mod edit -replace github.com/codeready-toolchain/toolchain-common=${C_PATH}
-    make verify-dependencies || ERRORLIST+="($(basename ${repo}))"
+    make verify-dependencies 2> >(tee ${err_file}) 1> >(tee ${stdout_file})
+    rc=$?
+    if [ ${rc} -ne 0 ]; then
+    ERRORREPOLIST+="$(basename ${repo}) " 
+    ERRORFILELIST+="${err_file}  "
+    STDOUTFILELIST+="${stdout_file} "
+    fi
     echo                                                          
     echo =========================================================================================
     echo                                                           
 done
-if [ ${#ERRORLIST[@]} -ne 0 ]; then
+echo                "Summary"
+if [ ${#ERRORREPOLIST[@]} -ne 0 ]; then
     echo "Below are the repos with error: "
-    for e in ${ERRORLIST[*]}
+    for errorreponame in ${ERRORREPOLIST[*]}
     do
-        echo "${e}"
+        for errorfilename in ${ERRORFILELIST[*]}
+        do
+            if [[ ${errorfilename} =~ ${errorreponame} ]]; then 
+                echo                                                          
+                echo =========================================================================================
+                echo 
+                echo                       "${errorreponame} has the following errors "
+                echo                                                          
+                echo =========================================================================================
+                echo 
+                cat "${errorfilename}"
+            else
+            # Since golint check is the only check for which we parse the error from the standard-ouput
+            # and is checked at last after all the other checks are done. But if there is any error in the previous checks,
+            # the script won't go ahead to check golint, and hence this check is seperated and put into else
+            # Meaning if the other checks have passed then only it wil proceed to check golint and we would parse 
+            # golint errors from std files if any.
+                for stdoutfilename in ${STDOUTFILELIST[*]}
+                do
+                    if [[ ${stdoutfilename} =~ ${errorreponame} ]]; then 
+                        cat "${stdoutfilename}" | grep -E ${GOLINTREGEX} 
+                    fi
+                done                                            
+            fi
+        done
```

Comment:
There is something weird with these nested for cycles.
There is one stderr file and one stdout file per repo, right?
The first for cycle goes through all error file names and tries to find the corresponding one for the failed repo. For every file whose name doesn't match the expected repo the code goes into the `else` part and starts a loop for all stdout file names trying to find matching one also there. 
But why doing it for every single time when the `errorfilename` doesn't match `errorreponame`? Why having the for loop nested inside the first one and as part of the `else` section? why not having the second loop at the same level as the first one?

Reply:
Okay the logic you suggested can also be one option, 
what i thought and as i wrote in the comments that, we get `go lint error` only if it hasn't failed in the previous checks (go-lint check is always last check) , and since i had to anyways go through a separate for loop i thought of checking only if there is no error file in the previous list then only go and check in the stdout file for go-lint errors

Reply:
Okay as we discussed, i would move it to the same level